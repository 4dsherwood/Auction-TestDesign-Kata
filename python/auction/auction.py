
class AuctionEventListener:
    # TODO: add methods here
    def notify(self, message):
        return message

class AuctionMessageTranslator:
    def __init__(self, listener: AuctionEventListener):
        self.listener = listener

    def process_message(self, message: str):
        if "CLOSE" in message:
            self.listener.notify("Auction Closed")

        elif "PRICE" in message:
            data = {}
            for element in message.split(";"):
                if not element:
                    continue
                pair = element.split(":")
                data[pair[0].strip()]  = pair[1].strip()

            current_price = int(data["CurrentPrice"])
            increment = int(data["Increment"])
            bidder = data["Bidder"]
            # 'Someone else bid 199'
            notification_message = f"{bidder} bid {current_price+increment}"

            self.listener.notify(notification_message)

        else:

            self.listener.notify("ERROR: Received a Blank Message")
            pass

