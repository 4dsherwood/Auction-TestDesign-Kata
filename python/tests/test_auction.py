import pytest
from approvaltests import verify

from auction.auction import AuctionEventListener, AuctionMessageTranslator

#from .auction import *   # E   ModuleNotFoundError: No module named 'tests.auction'
#from auction import *
#from AuctionEventListener import *


def test_notifies_auction_closed_when_close_message_received():
    message = "SOLVersion: 1.1; Event: CLOSE;"
    # TODO: write a test for this message translation

def test_notifies_auction_closed_when_close_message_received_using_mock():
    # Arrange: Set up the translator and mock listener
    class MockListener(AuctionEventListener):
        def __init__(self):
            self.notifications = []

        def notify(self,message):
            self.notifications.append(message)

    # Create an instance of the mock listener
    mock_listener = MockListener()

    # Create an instance of the message translator using the mock listener
    translator = AuctionMessageTranslator(mock_listener)

    # Act: Send the test message through the translator
    message = "SOLVersion: 1.1; Event: CLOSE;"
    translator.process_message(message)

    # Assert: Verify the output using ApprovalTests
    verify(mock_listener.notifications)


def test_notifies_bid_details_when_price_message_received():
    # Arrange: Set up the translator and mock listener
    class MockListener(AuctionEventListener):
        def __init__(self):
            self.notifications = []

        def notify(self,message):
            self.notifications.append(message)

    # Create an instance of the mock listener
    mock_listener = MockListener()

    # Create an instance of the message translator using the mock listener
    translator = AuctionMessageTranslator(mock_listener)

    # Act: Send the test message through the translator
    message = "SOLVersion: 1.1; Event: PRICE; CurrentPrice: 192; Increment: 7; Bidder: Someone else;"
    translator.process_message(message)

    # Assert: Verify the output using ApprovalTests
    verify(mock_listener.notifications)

def test_notifies_bid_details_when_price_message_received_blank():
    message = "SOLVersion: 1.1; Event: PRICE; CurrentPrice: 192; Increment: 7; Bidder: Someone else;"
    # TODO: write a test for this message translation
