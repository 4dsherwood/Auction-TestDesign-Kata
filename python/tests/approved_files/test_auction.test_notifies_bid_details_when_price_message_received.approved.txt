['Someone else bid 199']
